fastcrypto/src/groups/bls12381.rs

```diff
@@ -649,6 +650,22 @@ impl ScalarType for Scalar {
     }
 }
 
+impl HashToGroupElement for Scalar {
+    fn hash_to_group_element(msg: &[u8]) -> Self {
+        let hashed = Sha512::digest(msg).digest;
+        let mut ret = blst_fr::default();
+        let mut bytes = [0u8; SCALAR_LENGTH];
+        bytes.copy_from_slice(&hashed[..SCALAR_LENGTH]);
+        // TODO: is there an overflow?
+        unsafe {
+            let mut scalar = blst_scalar::default();
+            blst_scalar_from_bendian(&mut scalar, bytes.as_ptr());
+            blst_fr_from_scalar(&mut ret, &scalar);
```

Comment:
Since the modular reduction is done from a blst_scalar (which is 256 bits), the result will be biased.

Reply:
Replaced with FiatShamirChallenge